core/src/test/scala/akka/persistence/postgres/journal/PostgresJournalSpec.scala

```diff
@@ -54,6 +55,119 @@ abstract class PostgresJournalSpec(config: String, schemaType: SchemaType)
     super.afterAll()
   }
 
+  private def writeSingleMessage(seqNr: Int, pid: String, sender: ActorRef, writerUuid: String) = {
+    val msg = AtomicWrite(
+      PersistentRepr(
+        payload = s"a-$seqNr",
+        sequenceNr = seqNr,
+        persistenceId = pid,
+        sender = sender,
+        writerUuid = writerUuid))
+    val probe = TestProbe()
+    journal ! WriteMessages(List(msg), probe.ref, actorInstanceId)
+    probe.expectMsg(WriteMessagesSuccessful)
+  }
+
+  "A journal" must {
+    "not allow to store events with sequence number lower than what is already stored for the same persistence id" in {
+      // given
+      val perId = "perId"
+      val sender = TestProbe()
+      val repeatedSnr = 5
+
+      // when
+      writeMessages(1, repeatedSnr + 1, perId, sender.ref, writerUuid)
+
+      // then
+      val msg = AtomicWrite(
+        PersistentRepr(
+          payload = s"a-$repeatedSnr",
+          sequenceNr = repeatedSnr,
+          persistenceId = perId,
+          sender = sender.ref,
+          writerUuid = writerUuid))
+
+      val probe = TestProbe()
+      journal ! WriteMessages(Seq(msg), probe.ref, actorInstanceId)
+      probe.expectMsgType[WriteMessagesFailed]
+    }
+  }
+
+  "An insert on the journal" must {
+    import akka.persistence.postgres.db.ExtendedPostgresProfile.api._
+
+    val metadataTable = JournalMetadataTable(journalConfig.journalMetadataTableConfiguration)
+
+    "automatically insert journal metadata" in {
+      // given
+      val perId = "perId-meta-1"
+      val sender = TestProbe()
+
+      // when
+      writeSingleMessage(1, perId, sender.ref, writerUuid)
+
+      // then
+      val metadataExists = db.run(metadataTable.filter(_.persistenceId === perId).exists.result).futureValue
+      metadataExists shouldBe true
+    }
```

Comment:
@tiagomota Any way you could test the min ordering is correctly inserted in that case (when it's the first event of a persistence id) ?

Reply:
Yes. When it is the first insert for a persistence_id the `min_ordering` should be equal to the `max_ordering` (the journal `ordering` value is used in both columns), and it should be greater than 0.

I've added those checks to the test.